src/mtcp/mtcp_restart.c

```diff
@@ -1065,7 +1064,7 @@ adjust_for_smaller_file_size(Area *area, int fd)
     DPRINTF("For %s, current size (%ld) smaller than original (%ld).\n"
             "mmap()'ng the difference as anonymous.\n",
             area->name, curr_size, area->size);
-    VA mmappedat = mtcp_sys_mmap(anon_start_addr, anon_area_size,
+    VA mmappedat = mmap_fixed_noreplace(anon_start_addr, anon_area_size,
```

Comment:
REPEATED:
The `MAP_FIXED` is redundant, since `map_fixed_noreplace()` will simply remove the `MAP_FIXED`.
We can keep it here, or remove `MAP_FIXED` with the idea that the name `map_fied_noreplace()` already says it all.
I'm open to either option (keeping it as is, or removing it).  Just a thought.